### PR TITLE
SPDX vs NPM project-package name improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: plugins/package-managers/node/src/main/kotlin/Npm.kt [hunk 1]
```diff
@@ -534,8 +534,13 @@ open class Npm(
 
         val rawName = json["name"].textValueOrEmpty()
         val (namespace, name) = splitNpmNamespaceAndName(rawName)
-        if (name.isBlank()) {
-            logger.warn { "'$packageJson' does not define a name." }
+
```
Comment (fviernau) [inline plugins/package-managers/node/src/main/kotlin/Npm.kt:537]: commit-message: like to one -> like the one

File: plugins/package-managers/node/src/main/kotlin/Npm.kt [hunk 2]
```diff
@@ -534,8 +534,13 @@ open class Npm(
...
+        val projectName = if (name.isBlank()) {
+            getFallbackProjectName(analysisRoot, packageJson).also {
+                logger.warn { "'$packageJson' does not define a name, falling back to '$it'." }
```
Comment (fviernau) [inline plugins/package-managers/node/src/main/kotlin/Npm.kt:540]: We have a couple of callers of `getFallbackProjectName()`, but none logs any message about the use of the fallback. What doy ou think about simply dropping this here as well?

Comment (fviernau) [reply plugins/package-managers/node/src/main/kotlin/Npm.kt:540]: Then, code can be shortened, e.g. to:
`= name.takeUnless { it.isBlank() } ?: getFallbackProjectName(analysisRoot, packageJson)`

**Review by fviernau — CHANGES_REQUESTED** [review]

File: plugins/package-managers/node/src/main/kotlin/Npm.kt (hunk 2 above)
Comment (sschuberth) [reply plugins/package-managers/node/src/main/kotlin/Npm.kt:540]: > What doy ou think about simply dropping this here as well?

This is basically just the previously existing log statement, extended with the fallback name. It did not feel right to me to remove that log statement altogether in this context, esp. as there is a similar log statement about the version, too, which should then also be removed for consistency.

Comment (sschuberth) [reply plugins/package-managers/node/src/main/kotlin/Npm.kt:540]: However, I've simplified the code via `ifBlank {}` now.

**Review by fviernau — APPROVED** [review]

**Pull request merged**